### PR TITLE
State: Introduce a getApplicationPasswords selector

### PR DESCRIPTION
**Pull request by tyxla**
This PR is part of #22912 and introduces a simple selector to fetch the application passwords.

To test, verify the tests pass:

```
npm run test-client client/state/selectors/test/get-application-passwords.js
```

### CODE REVIEW

File: client/state/selectors/test/get-application-passwords.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+				value: '2018-02-01T08:10:00+00:00',
+			},
+		];
+		const state = {
+			applicationPasswords: {
+				items: appPasswords,
+			},
+		};
+		const result = getApplicationPasswords( state );
+		expect( result ).toEqual( appPasswords );
```
Comment (sirreal) [inline client/state/selectors/test/get-application-passwords.js]: Could be `toBe`.

File: client/state/selectors/test/get-application-passwords.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	} );
+
+	test( 'should return an empty array if no application passwords exist', () => {
+		const state = {
+			applicationPasswords: {
+				items: [],
+			},
+		};
+		const result = getApplicationPasswords( state );
+		expect( result ).toEqual( [] );
```
Comment (sirreal) [inline client/state/selectors/test/get-application-passwords.js]: Does this test add any value? It's testing the same behavior as the previous test, isn't it?

Comment (tyxla) [reply client/state/selectors/test/get-application-passwords.js]: You're right - removed in https://github.com/Automattic/wp-calypso/pull/23906/commits/787c87f0b9618bfa9f7277783108a0aeb66b4b09.

File: client/state/selectors/test/get-application-passwords.js (hunk 1 above)
Comment (tyxla) [reply client/state/selectors/test/get-application-passwords.js]: https://github.com/Automattic/wp-calypso/pull/23906/commits/e41f8c698b858a3379b035541852db9db4f950a7

**Pull request merged**